File: algorithm/python/BOJ11650.py
# problem : https://www.acmicpc.net/problem/11650

import sys


N = int(sys.stdin.readline().rstrip())
point_list = list()

for i in range(0, N):
    x, y = map(int, sys.stdin.readline().rstrip().split())
    point_list.append([x, y])

point_list.sort(key=lambda p: (p[0], p[1]))

for point in point_list:
    print(point[0], point[1])

File: algorithm/typescript/BOJ11650.ts
// problem : https://www.acmicpc.net/problem/11650

import fs = require("fs");

class Point {
  x: number;
  y: number;

  constructor(x: number, y: number) {
    this.x = x;
    this.y = y;
  }
}

const input: string[] = fs.readFileSync("/dev/stdin").toString().split("\n");
const N = Number(input[0]);
const points: Point[] = [];

for (let i = 1; i <= N; i++) {
  const inputData: number[] = input[i].split(" ").map(Number);
  const p = new Point(Number(inputData[0]), Number(inputData[1]));
  points.push(p);
}

points.sort((p1: Point, p2: Point) => {
  let value = p1.x - p2.x;

  if (value === 0) {
    value = p1.y - p2.y;
  }

  return value;
});

console.log(points.map((point) => `${point.x} ${point.y}`).join("\n"));
